### PR TITLE
root development refactor

### PR DESCRIPTION
**Pull request by morrowcj**
This pull request involves the refactoring of the crop root development processes. It is a small PR and involves only a handful of functions.

The main change in this PR is [root_development.py](SC_redesign/Crop_and_Soil/crop/root_development.py) and associated tests: [test_root_development.py](SC_redesign/Crop_and_Soil/tests_SC/test_root_development.py). With this PR, I've used the pattern suggested to me by @ltp23, which I think works well. 

This PR follows #224. Review of this PR should probably wait until #224 is merged into SC_redesign.

### CODE REVIEW

File: SC_redesign/Crop_and_Soil/crop/crop.py [hunk 1]
```diff
@@ -4,18 +4,25 @@
 from SC_redesign.Crop_and_Soil.crop.water_dynamics import WaterDynamics
 from SC_redesign.Crop_and_Soil.crop.heat_units import HeatUnits
 from SC_redesign.Crop_and_Soil.crop.leaf_area_index import LeafAreaIndex
-from SC_redesign.Crop_and_Soil.soil.soil import Soil
+from SC_redesign.Crop_and_Soil.crop.root_development import RootDevelopment
+
+from typing import List
 
 # TODO: Should use an ENUM class to represent the supported species??
```
Comment (Isgotkowitz) [inline SC_redesign/Crop_and_Soil/crop/crop.py:11]: Sounds like a good idea, but this require extra code to update the ENUM every time a new crop species is added right?

File: SC_redesign/Crop_and_Soil/crop/root_development.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            return 0.4 - 0.2 * heat_fraction
+
+    @staticmethod
+    def determine_root_depth(max_depth: float, heat_fraction: float) -> float:
+        """calculates a plant's root depth on a given day
+
+        Args:
+            max_depth: maximum possible root depth (mm)
+            heat_fraction: fraction of potential heat units
+
```
Comment (Isgotkowitz) [inline SC_redesign/Crop_and_Soil/crop/root_development.py:52]:
```suggestion

SWAT Reference: 5:2.1.23, 24

```

**Review by Isgotkowitz — COMMENTED** [review]
Adding the SWAT reference in `root_development.py` is the only thing I think should be changed, otherwise LGTM

File: SC_redesign/Crop_and_Soil/crop/crop.py (hunk 1 above)
Comment (morrowcj) [reply SC_redesign/Crop_and_Soil/crop/crop.py:11]: I'm not sure. I don't fully understand ENUMs yet. It is something I want to learn more about (perhaps you can enlighten me a bit) before moving forward. I did a small amount of reading on the pattern and it seemed useful for what I'm planning. We'll need to re-evaluate later when we start working on the details of the `Crop` class.

File: SC_redesign/Crop_and_Soil/crop/root_development.py (hunk 2 above)
Comment (morrowcj) [reply SC_redesign/Crop_and_Soil/crop/root_development.py:52]: Good catch, thanks!

**Review by Isgotkowitz — APPROVED** [review]

File: SC_redesign/Crop_and_Soil/crop/crop.py [hunk 3]
```diff
@@ -57,6 +62,13 @@ def grow_crop(self, layer_nitrates: list[float], layer_depths: list[float], soil
         self.allocate_biomass(incoming_light)
         self.cycle_water(evaporation, transpiration, max_evapotranspiration)
 
+    def list_all_var_names(self):
```
Comment (Isgotkowitz) [inline SC_redesign/Crop_and_Soil/crop/crop.py]: This function needs a return type

File: SC_redesign/Crop_and_Soil/crop/crop.py (hunk 1 above)
Comment (Isgotkowitz) [reply SC_redesign/Crop_and_Soil/crop/crop.py:11]: Ok, I'm not sure how ENUMs work in Python so I will have to do some learning too.

**Review by Isgotkowitz — CHANGES_REQUESTED** [review]

File: SC_redesign/Crop_and_Soil/crop/crop.py (hunk 3 above)
Comment (morrowcj) [reply SC_redesign/Crop_and_Soil/crop/crop.py]: Done. Thanks

**Review by Isgotkowitz — APPROVED** [review]

File: SC_redesign/Crop_and_Soil/crop/crop.py [hunk 4]
```diff
@@ -57,6 +62,13 @@ def grow_crop(self, layer_nitrates: list[float], layer_depths: list[float], soil
         self.allocate_biomass(incoming_light)
         self.cycle_water(evaporation, transpiration, max_evapotranspiration)
 
+    def list_all_var_names(self) -> list:
+        """list all variables used by Crop"""  # TODO: check for duplicates or conflicts among parents
+        return vars(self)
```
Comment (ltp23) [inline SC_redesign/Crop_and_Soil/crop/crop.py]: I believe `vars(obj)` returns a dictionary, so if you want a list, then use `list(vars(obj).keys())` or `list(vars(obj).values())` depending on what makes sense to you.

**Review by ltp23 — APPROVED** [review]
Everything looks great! Thanks for your work.

File: SC_redesign/Crop_and_Soil/crop/crop.py (hunk 4 above)
Comment (morrowcj) [reply SC_redesign/Crop_and_Soil/crop/crop.py]: No, I think returning a dictionary makes more sense. I'll fix it and then merge.

**Pull request merged**